nada_numpy/array.py

```diff
@@ -390,6 +390,106 @@ def __imatmul__(self, other: Any) -> "NadaArray":
         """
         return self.matmul(other)
 
+    def __comparison_operator(self, value: Any, operator: Callable) -> "NadaArray":
+        """
+        Perform element-wise comparison with broadcasting.
+
+        NOTE: Specially for __eq__ and __ne__ operators, the result expected is bool.
```

Comment:
Is this the expected result? In numpy, __eq__ and __ne__ are broadcasted
I believe this implementation matches that behaviour, no?

Reply:
Yes, in NumPy, those two operations are indeed broadcasted. However, there's an unusual behavior when you try to do `array_a == array_b`. NumPy evaluates the boolean value of each element, as these operations are expected to result in either `True` or `False`. For instance, it will determine whether a `SecretInteger` is `True` or `False`—which always turns out to be `True`.